### ISSUE TITLE
Upgrade D3 dependency

### ISSUE THREAD

**Issue by vilim** [idyll-lang/idyll#754]
Most d3 documentation available online is on newer versions of D3 (currently at 7.1), while the one bundled with Idyll is 4.0 (the changes I noticed concerned handling interactions like `.drag` though I am sure there are many more). Trying to upgrade d3 only via npm results in an error on building the document:

```
Error [ERR_REQUIRE_ESM]: require() of ES Module .../node_modules/d3/src/index.js from /home/vilimstih/Vara/interactive/rocs/components/custom-d3-component.js not supported.
Instead change the require of index.js in .../components/custom-d3-component.js to a dynamic import() which is available in all CommonJS modules.
```
Changing the `require` to `import` however results in the d3 code not working at all (`.select` cannot be found as a function)

**Comment by vilim** [comment]
It turns out not to be high-priority as D3 version 6 works nicely and the documentation is valid for this version.

**Comment by mathisonian** [comment]
Thanks for flagging @vilim - using v6 is recommended for now and I've opened a PR to use that version by default instead of the rather old v4 that is being used now. 

v7 support requires a slightly bigger change to support ES modules - this is in the works but I don't expect it to be finished until sometime in January. I'll leave this open until then.

**Comment by mathisonian** [comment]
Working on a solution in https://github.com/idyll-lang/idyll/pull/764

**Comment by mathisonian** [comment]
This is fixed in the latest version of idyll, currently released as an alpha version on npm `idyll@6.0.0-alpha.5`
